lib/logdna.rb

```diff
@@ -20,50 +20,24 @@ def initialize(key, opts={})
       @level = opts[:level] || 'INFO'
       @env = opts[:env]
       @meta = opts[:meta]
-      @@client = nil unless defined? @@client
-      
+
       endpoint = opts[:endpoint] || Resources::ENDPOINT
       hostname = opts[:hostname] || Socket.gethostname
       ip =  opts.key?(:ip) ? "&ip=#{opts[:ip]}" : ''
       mac = opts.key?(:mac) ? "&mac=#{opts[:mac]}" : ''
       url = "#{endpoint}?hostname=#{hostname}#{mac}#{ip}"
+      uri = URI(url)
 
-      begin
-        if (hostname.size > Resources::MAX_INPUT_LENGTH || @app.size > Resources::MAX_INPUT_LENGTH )
-            raise MaxLengthExceeded.new
-        end
-      rescue MaxLengthExceeded => e
+      if (hostname.size > Resources::MAX_INPUT_LENGTH || @app.size > Resources::MAX_INPUT_LENGTH )
         puts "Hostname or Appname is over #{Resources::MAX_INPUT_LENGTH} characters"
-        handle_exception(e)
-        return
       end
```

Comment:
This used to early return and doesn't anymore - is that intentional?

Reply:
The whole logic here is weird. Why to catch an exception that they throw by themselves ? =)

Reply:
And I am not even sure how useful this check is.  Do we care so much about how long the hostname they use? Why do we care about this only in Ruby code? Why don't we care then about the length of the app name, level name etc...

Reply:
Although I don't have any limitation to the message length. I can add it.

Reply:
> The whole logic here is weird. Why to catch an exception that they throw by themselves ? =)

To log / print the exception. It's a little janky, but we should figure out why we weren't setting up a client in this hostname/app size case before. I get you have questions - we should get answers.

Reply:
Actually we have the checks for all the parameters sent in options in nodejs lib. I'll add them here as well.